scripts/coverage_report_jenkins.py

```diff
@@ -51,6 +123,218 @@ def get_build_numbers(client, job_name):
     return [build['number'] for build in client.get_job_info(job_name)['builds']]
 
 
+def merge_coverage_data(ssh, coverage_dir):
+    """Merge coverage data
+
+    Take all the by appliance by process .resultset.json files from
+    the coverage archive and merge them into one .resultset.json file.
+    Expects the coverage archive to have been extracted to the
+    coverage_dir on the appliance to which the ssh client is connected.
+
+    Args:
+        ssh:  ssh client
+        coverage_dir:  Directory where the coverage archive was extracted.
+
+    Returns:
+        Nothing
+    """
+    logger.info('Merging coverage data')
+
+    # Run the coverage merger script out of the rails root pointing
+    # to where the coverage data is installed.   This will generate
+    # a directory under the coverage directory called merged, and
+    # will have the merged .resultset.json file in it, along with some
+    # HTML that was generated by the merger script.
+    cmd = ssh.run_rails_command(
+        'coverage_merger.rb --coverageRoot={}'.format(coverage_dir),
+        timeout=60 * 60)
+    if cmd.failed:
+        raise Exception('Failure running the merger - {}'.format(str(cmd)))
+
+    # Attempt to get the overall code coverage percentage from the result.
+    logger.info('Coverage report generation was successful')
+    logger.info(str(cmd))
+    percentage = re.search(r'LOC\s+\((\d+.\d+%)\)\s+covered\.', str(cmd))
+    if percentage:
+        logger.info('COVERAGE={};'.format(percentage.groups()[0]))
```

Comment:
@jamesooden You need to update all of these calls as @quarckster pointed out on one of them. See the last line of my comment from the original suggestion too - logger takes vars for formatting as arguments, and `.format()` doesn't need to be used in this context..

https://github.com/ManageIQ/integration_tests/pull/6705/files#r171052902